src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,95 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Encapsulates the idea of tasks that need to be done by a certain date and time
+ */
+public class Deadline extends Task {
+
+    // constants
+    private static final String TYPE = "D";
+    private static final DateTimeFormatter DATABASE_DATE_TIME_FORMAT = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+
+    // instance variables
+    private LocalDateTime deadline;
+
+    // constructors
+
+    /**
+     * Constructs a Deadline object without a date
+     *
+     * @param description the description of the Deadline
+     */
+    public Deadline(String description) {
+        super(description);
+    }
+
+    /**
+     * Constructs a Deadline object with the given due date
+     *
+     * @param description the description of the Deadline
+     * @param due         the date of the deadline
+     */
+    public Deadline(String description, String due) {
+        super(description);
+        this.deadline = LocalDateTime.parse(due, DATABASE_DATE_TIME_FORMAT);
+    }
+
+    /**
+     * Constructs a Deadline object with the given due date and whether it is completed
+     *
+     * @param description the description of the Deadline
+     * @param deadline    the date of the deadline
+     * @param isDone      whether the task is completed
+     */
+    public Deadline(String description, String deadline, boolean isDone) {
+        super(description, isDone);
+        this.deadline = LocalDateTime.parse(deadline, DATABASE_DATE_TIME_FORMAT);
+    }
+
+    /**
+     * Helper method to format the date according to the
```

Comment:
I think that there isn't a need to indicate that it is a helper method.

Perhaps this would be a better start to your Javadoc. 
```suggestion
     * Formats the date according to the
```

I noticed the same issue at one other place too.